### ISSUE TITLE
[Question] How do I get a custom field out of the payload

### ISSUE THREAD

**Issue by milesje** [jwtk/jjwt#230]
OK, I am adding a couple of custom claims to the payload when I generate the JWT, and I can pull those out just fine in my front-end (javascript) just fine. I then have my javascript send an ajax call to a micro-service and it passes the JWT along with it. I want to get my custom claims out of the JWT in the micro-service. I'm doing the following:

```
 Claims claims = Jwts.parser().setSigningKey(Vars.SECRET_KEY).parseClaimsJws(token).getBody();
 User user = claims.get("customuser", User.class);
```

and it throws an exception.
```
io.jsonwebtoken.RequiredTypeException: Expected value to be of type: class net.netdatacorp.netdauth.model.User, but was class java.util.LinkedHashMap
	at io.jsonwebtoken.impl.DefaultClaims.get(DefaultClaims.java:128)
```

Here is how the data looks in the JWT inspector on the front-end for my custom claim.
```
{
  jti: "83bffbad-7d36-4370-9332-21a84f2a3dce",
  iat: 1498241526,
  sub: "test",
  iss: "www.test.net",
  customuser: {
    userId: 1,
    userCd: "TMM",
    firstNm: "Testy",
    lastNm: "McTesty",
    userNm: "test",
    emailAddress: "jacob@test.net",
    active: true,
    createdDt: 1491355712000,
    createdByUserId: 0,
    lastUpdateDt: 1498199278000,
    lastUpdateByUserId: 0,
    lastLoginDt: 1484928016000
  }
}
```

What am I missing to be able to pull my custom claim out?

**Comment by milesje** [comment]
Moved this to StackOverflow.

**Comment by gempain** [comment]
@milesje Can you reference the SO question ?

**Comment by gempain** [comment]
For reference, here is the SO question: https://stackoverflow.com/questions/44728138/how-do-i-get-a-custom-field-out-of-the-payload-using-jjwt

**Comment by slebetman** [comment]
So the answer to this question is to stop using this library and use Jose4J instead? Great!

**Comment by bdemers** [comment]
@slebetman take a look at: https://github.com/jwtk/jjwt#parsing-of-custom-claim-types
If that doesn't help, feel free to open another issue asking for doc clarification

**Comment by lhazlewood** [comment]
JJWT has this functionality as documented in @bdemers link - it was released in 0.11.0.

The reason why JJWT does it this particular way (through the JSON marshaller) is that is because marshaling behavior (converting to/from JSON and custom object types) is squarely in the domain/responsibility of JSON marshaller libraries (like Jackson, etc).  There's no need for JJWT to reinvent this wheel when the marshaller can do it automatically when parsing the raw JSON content.  HTH!

**Comment by lhazlewood** [comment]
I just added an answer to the SO question (I hadn't seen it until now), so hopefully this helps others in the future (please upvote if it satisfies your needs: https://stackoverflow.com/a/66433213/407170 ).  Thanks for the comment!
